pandas/parser.pyx

```diff
@@ -662,7 +662,11 @@ cdef class TextReader:
 
         if isinstance(source, basestring):
             if not isinstance(source, bytes):
-                source = source.encode(sys.getfilesystemencoding() or 'utf-8')
+                if compat.PY36 and compat.is_platform_windows():
+                    fs_encoding = 'mbcs'
```

Comment:
pls move these changes to io/common.py as this needs to be used in several places. make a function called ``_get_filename_encoding``